arviz/plots/lmplot.py

```diff
@@ -0,0 +1,331 @@
+"""Plot linear regression figure."""
+import warnings
+from numbers import Integral
+
+import xarray as xr
+import numpy as np
+from xarray.core.dataarray import DataArray
+
+from ..sel_utils import xarray_var_iter
+from ..rcparams import rcParams
+from .plot_utils import default_grid, filter_plotters_list, get_plotting_function
+
+
+def plot_lm(
+    y,
+    idata=None,
+    x=None,
+    y_model=None,
+    y_hat=None,
+    num_pp_samples=50,
+    kind_pp="samples",
+    kind_model="lines",
+    xjitter=False,
+    plot_dim=None,
+    backend=None,
+    y_kwargs=None,
+    y_hat_plot_kwargs=None,
+    y_hat_fill_kwargs=None,
+    y_model_plot_kwargs=None,
+    y_model_fill_kwargs=None,
+    backend_kwargs=None,
+    show=None,
+    figsize=None,
+    textsize=None,
+    axes=None,
+    legend=True,
+    grid=True,
+):
+    """Plot linear regression figure to visualize uncertaininty in mean and ppc values.
+
+    Parameters
+    ----------
+    y : str or DataArray or ndarray
+        If str, variable name from observed_data
+    idata : az.InferenceData object, Optional
```

Comment:
I think it should be `InferenceData` alone, or alternatively `arviz.InferenceData`, object is not necessary, and I think `az` can be confusing and redundant (arviz.InferenceData is as redundant, but IIUC, as arviz is the valid package name it will be rendered as a link to inferencedata api docs)